resources/qml/Sidebar.qml

```diff
@@ -256,6 +256,44 @@ Rectangle
         }
     }
 
+    Button
+    {
+        id: toggleFilterButton
+
+        anchors.right: parent.right
+        anchors.rightMargin: UM.Theme.getSize("default_margin").width
+        anchors.top: headerSeparator.bottom
+        anchors.topMargin: UM.Theme.getSize("default_margin").height
+
+        height: settingsModeSelection.height
+        width: visible ? height : 0
```

Comment:
This is used so the "advanced" button expands to its original width when the user is in "simple" mode. In "simple" mode the toggle that is only relevant in the "advanced" mode is simply hidden altogether to avoid confusion and to maintain simplicity. When the user switches to "advanced" mode the "advanced" button is resized slightly to make room for the toggle button.